src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,16 +35,121 @@
 
   /*
    * Called when a client submits a GET request to the /data URL
-   * Displays all recorded user comments on page
    */
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {}
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // PROTO Data structure:
+    // Parse the contents  of graph.txt into a proto Graph object, and extract information
+    // from the proto object into a map. This is used to store the proto input and isn't updated
+    // with mutations.
+    Graph protoGraph =
+        Graph.parseFrom(getServletContext().getResourceAsStream("/WEB-INF/graph.txt"));
+    Map<String, Node> protoNodesMap = protoGraph.getNodesMapMap();
+
+    // GRAPH Data structures:
+    // Create an undirected graph data structure to store the information, and
+    // map each node name in the graph to the GraphNode objects. This is the graph & map
+    // we update with mutations
+    MutableGraph<GraphNode> graph = GraphBuilder.directed().build();
+    HashMap<String, GraphNode> graphNodesMap = new HashMap<>();
+
+    // Create a graph (useable form) from proto data structures
+    for (String nodeName : protoNodesMap.keySet()) {
```

Comment:
I think this for loop could live in its own function. You should also write some unit tests for this function:
 - create a protoNodesMap
 - call this function
 - assert properties about the graph, such as nodes in the graph and edges in the graph